### ISSUE TITLE
online training

### ISSUE THREAD

**Issue by zingomaster** [ZhengyaoJiang/PGPortfolio#39]
Hello 

Thanks for the wonderful work, i read your paper and almost studied most of the code. However, i don't get the concept of append_experience and agent train in the rolling_train method 
I have some questions if i may ask 
1- what is the format of the saved experience and how does it affect the model ?
2- how is that different from training the model directly using self._agent.train() ? 
3- is the experience mentioned here the same as the mini-batches mentioned in the paper for online learning section 5.3 for example ? 

thanks in advance
Sarah Ahmed

**Comment by ZhengyaoJiang** [comment]
> what is the format of the saved experience 

It's just the history price movement saved at `self.__globaldata`

> how is that different from training the model directly using self._agent.train()

The only difference is the rolling train will append new history from the test set to the training set, during backtest.

**Comment by zingomaster** [comment]
Thank you for your reply 

> The only difference is the rolling train will append new history from the test set to the training set, during backtest.

I don't understand this part clearly (the experience appending part), where is this process actually happening ? i see the append experience appends index of a training data , but how get next batch returns the last omega ? where and when is the last omega saved ?

**Comment by ZhengyaoJiang** [comment]
> where and when is the last omega saved ?

`self.__PVM`, PVM is short for portfolio vector memory.

**Comment by zingomaster** [comment]
Thank you. 

I have 1 last question please, sorry for being so much demanding.
what is the online_w in datamatrices.append_experience(self, online_w=None) ? why do we need it ? 

Regards,

**Comment by DexHunter** [comment]
> what is the online_w in datamatrices.append_experience(self, online_w=None) ? why do we need it ?

I think `online_w` is mostly for live trading. You can probably ignore that in backtest

**Comment by zingomaster** [comment]
i know it's for live trading, i meant what is it ? i don't see anything telling what is it either in the paper or in the code

**Comment by DexHunter** [comment]
basically it's previous portfolio weight

**Comment by zingomaster** [comment]
okaaay, thank you a lot :) :100:
